fix: add retry logic for secureblue-flatpak-setup.service

HastD opened the pull request:
The retry logic is copied from securebluefirstrun.service and restarts the service on failure, with exponential backoff in the event of repeated failures.

RoyalOughtness commented on files/scripts/enablecommonautoupdate.sh:
```diff
@@ -17,4 +17,4 @@ set -oue pipefail
 systemctl --global enable podman-auto-update.timer
-systemctl --global enable secureblue-flatpak-setup.timer
\ No newline at end of file
+systemctl --global enable secureblue-flatpak-setup.service
```
using the timer is correct, the timer has the logic for triggering the service

RoyalOughtness commented on files/system/usr/lib/systemd/user/secureblue-flatpak-setup.service:
```diff
@@ -8,3 +8,10 @@ ConditionPathExists=!%h/.config/secureblue/secureblue-flatpak-setup.stamp
 ExecStartPost=/bin/bash -c 'mkdir -p "$HOME/.config/secureblue" && touch "$HOME/.config/secureblue/secureblue-flatpak-setup.stamp"'
+Restart=on-failure
+RestartSec=30s
+RestartMaxDelaySec=1800s
```
Oh, this could work also.

RoyalOughtness commented on .github/workflows/integration_tests/check_flatpak_setup.sh:
```diff
@@ -0,0 +1,56 @@
+
+check-flatpak-remotes() {
+    if ! grep -q 'flathub-verified' <(flatpak remotes); then
+        echo "Fail: flathub-verified flatpak remote not present."
```
this should be slightly more thorough in that flathub-verified should be the only remote, not just present

HastD replied:
Good point, I'll fix that.

HastD committed:
fix: add retry logic for secureblue-flatpak-setup.service

The retry logic is copied from securebluefirstrun.service and restarts
the service on failure, with exponential backoff in the event of
repeated failures.

Also add integration test for the service, and skip flatpak checks in
the audit service since that could check the flatpak setup service in
the middle of running and give timing-dependent results.

Signed-off-by: Daniel Hast <hast.daniel@protonmail.com>
